src/hb-ot-math-table.hh

```diff
@@ -1104,6 +1104,23 @@ struct MATH
 		  mathVariants.sanitize (c, this));
   }
 
+  // https://github.com/harfbuzz/harfbuzz/issues/4653
+  HB_INTERNAL bool is_bad_cambria (hb_font_t *font) const
+  {
+    hb::unique_ptr<hb_blob_t> table_blob {font->face->reference_table (tableTag)};
+    switch HB_CODEPOINT_ENCODE3 (table_blob->length,
+                                 get_constant (HB_OT_MATH_CONSTANT_DISPLAY_OPERATOR_MIN_HEIGHT, font),
+                                 get_constant (HB_OT_MATH_CONSTANT_DELIMITED_SUB_FORMULA_MIN_HEIGHT, font))
+    {
+      /* sha1sum:ab4a4fe054d23061f3c039493d6f665cfda2ecf5  cambria.ttc
+       * sha1sum:086855301bff644f9d8827b88491fcf73a6d4cb9  cambria.ttc
+       * sha1sum:b1e5a3feaca2ea3dfcf79ccb377de749ecf60343  cambria.ttc */
```

Comment:
So have you checked other popular OpenType MATH fonts have different sha1sums?

Using the length of the MATH table sounds good, but I wonder if the two constants are the best way to fingerprint Cambria Math? It seems likely that other fonts could have same values, especially if font designers that just copy what Cambria Math does...

Maybe OT::GDEF::is_blocklisted seems to use other table lengths instead, maybe we could do something similar here?

Or is the idea to rely them on these math constants so that the workaround is ignored if Microsoft fixes that in the future ? (unclear what their plan is)

Reply:
These checksums are here to just document what font files were checked. We check the table length, plus the values of the two constants, I found it quite unlikely that a different font will have exactly the same three values. I didn’t use the lengths of other tables because they seem irrelevant to MATH table, and an update to Cambria is likely to, say, change GSUB or GPOS while MATH table is unchanged, so we will need to add more entries to each version of Cambria now and in the future until it gets fixed.

Reply:
I mean, my question was have you tried with other popular fonts like STIX Two MATH, Latin Modern MATH etc and checked they don't return true for is_bad_cambria? I agree it's unlikely to happen, but I think it's worth checking, to avoid more breakage...

Reply:
If you can't get the same length, it seems you could still play with the second value overflow, making it overflow the 21 bits: https://searchfox.org/mozilla-central/rev/ca0f952fed01dc503a9cac23759da68ed137b337/gfx/harfbuzz/src/hb-algs.hh#216